src/main/java/bio/terra/cli/apps/CommandRunner.java

```diff
@@ -0,0 +1,136 @@
+
+import static bio.terra.cli.service.WorkspaceManager.getAiNotebookInstanceName;
+import static bio.terra.cli.service.WorkspaceManager.getBigQueryDatasetPath;
+import static bio.terra.cli.service.WorkspaceManager.getGcsBucketUrl;
+
+import bio.terra.cli.command.exception.SystemException;
+import bio.terra.cli.context.GlobalContext;
+import bio.terra.cli.context.WorkspaceContext;
+import bio.terra.workspace.model.ResourceDescription;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class CommandRunner {
+  private static final Logger logger = LoggerFactory.getLogger(CommandRunner.class);
+
+  protected final GlobalContext globalContext;
+  protected final WorkspaceContext workspaceContext;
+
+  public CommandRunner(GlobalContext globalContext, WorkspaceContext workspaceContext) {
+    this.globalContext = globalContext;
+    this.workspaceContext = workspaceContext;
+  }
+
+  /**
+   * Utility method for concatenating a command and its arguments.
+   *
+   * @param command the command and arguments (e.g. {gsutil, ls, gs://my-bucket})
+   */
+  protected static String buildFullCommand(List<String> command) {
+    String fullCommand = "";
+    if (command != null && command.size() > 0) {
+      final String argSeparator = " ";
+      fullCommand += argSeparator + String.join(argSeparator, command);
+    }
+    return fullCommand;
+  }
+
+  /**
+   * Run a tool command. Passes global and workspace context information as environment variables:
+   * pet SA key file, workspace GCP project, resolved workspace resources.
+   *
+   * @param command the command and arguments to execute
+   */
+  public void runToolCommand(List<String> command) {
+    runToolCommand(command, new HashMap<>());
+  }
+
+  /**
+   * Run a tool command. Passes global and workspace context information as environment variables:
+   * pet SA key file, workspace GCP project, resolved workspace resources. Allows adding environment
+   * variables beyond this, as long as the names don't conflict.
+   *
+   * @param command the command and arguments to execute
+   * @param envVars a mapping of environment variable names to values
+   * @throws SystemException if a Terra environment variable overlaps or conflicts with one passed
+   *     into this method
+   */
+  public void runToolCommand(List<String> command, Map<String, String> envVars) {
+    for (String commandToken : command) {
+      logger.debug("tokenized command string: {}", commandToken);
+    }
+
+    // check that the current workspace is defined
+    workspaceContext.requireCurrentWorkspace();
+
+    // add Terra global and workspace context information as environment variables
+    Map<String, String> terraEnvVars = buildMapOfTerraReferences();
+    terraEnvVars.put(
+        "GOOGLE_APPLICATION_CREDENTIALS",
+        GlobalContext.getPetSaKeyFile(globalContext.requireCurrentTerraUser(), workspaceContext)
```

Comment:
Moved this line to the `LocalProcessCommandRunner` class, since that's the only place that's using this particular value. But I kept the check here for any collisions.